### ISSUE TITLE
Query against a connection that has dropped results in an unhandled promise rejection

### ISSUE THREAD

**Issue by Hexxeh** [typeorm/typeorm#767]
```
async connect() {
  this.db = await TypeORM.createConnection({
    ...
  });
}

healthcheck() {
  return this.db.query('SELECT 1;');
}
```

In the example code above, if the MySQL server goes away whilst the application is running, the promise returned by healthcheck function will never resolve or reject and an unhandled rejection is raised like so: "connect ECONNREFUSED 127.0.0.1:3306".

It should be possible to handle the dropped connection and should not result in a never-resolving promise.

**Comment by pleerock** [comment]
Test to reproduce this issue:

```ts
createConnection({
            "type": "mysql",
            "host": "localhost",
            "port": 3306,
            "username": "test",
            "password": "test",
            "database": "test"
        }).then(connection => {
            console.log("connected, close mysql in 10 seconds...");
            setTimeout(async () => {
                try {
                    await connection.query("SELECT * FROM post");
                } catch (err) {
                    console.log("err in test: ", err);
                }
                ok();
            }, 10000);
        });
```

**Comment by pleerock** [comment]
Fixed and will be released in `0.1.0-alpha.38`
